### DISCUSSION TITLE
Customised commands used in tblr environment

### DISCUSSION THREAD

**Discussion by cdacl** [lvjr/tabularray#43] [Q&A]
I want to type several cells in a row repeatly within different tables, like this:

`| grade 1 | grade 2 | grade 3 | grade 4 | grade 5 |`

so I defined a command with  the context above:

`\newcommand\test{grade 1 & grade 2 & grade 3 & grade 4 & grade 5}`

I used this with Tabularx, 

```
\newcommand\test{grade 1 & grade 2 & grade 3 & grade 4 & grade 5}

The \emph{Tabularx} output: 

\begin{tabular}{|c|c|c|c|c|}
\hline
\test \\
\hline
one & two & three & four & five \\
\hline
\end{tabular}
``` 
and output was ok.
![1](https://user-images.githubusercontent.com/87848848/126747169-a591576c-ce6e-48f2-b7b4-55fdb83ac231.png)

I used this in tblr, but the output was not as I expected. The codes:
```
The \emph{tblr} output with $\backslash$newcommand:

\begin{tblr}{|c|c|c|c|c|}
\hline
\test \\
\hline
one & two & three & four & five \\
\hline
\end{tblr}
```
The output:
![2](https://user-images.githubusercontent.com/87848848/126747636-2f1c1d5d-f4cc-49ce-bae1-b54d959e270d.png)

I changed the newcommand with NewTableCommand, It's not as expected, too. The codes:
```
\NewTableCommand\testtblr{grade 1 & grade 2 & grade 3 & grade 4 & grade 5}
The \emph{tblr} output with $\backslash$NewTableCommand:

\begin{tblr}{|c|c|c|c|c|}
\hline
\testtblr \\
\hline
one & two & three & four & five \\
\hline
\end{tblr}
```
The output:
![3](https://user-images.githubusercontent.com/87848848/126748005-94b69099-0335-43f1-ad0c-a91894cde5ea.png)

How can I make it works like Tabularx package?

Thanks.

**Comment by lvjr** [comment]
You can not put cell text inside `\NewTableCommand`. In the latest source code (which will be in the next  release planned on 1st August) there is an `expand` option for expanding command once. See #18 and #23.

**Reply by cdacl** [reply]
I got it. Thank you very much.

**Comment by lvjr** [comment]
Workarounds for current version:

https://tex.stackexchange.com/questions/604228/tabularray-reading-a-table-stored-in-a-macro-by-catchfile

https://tex.stackexchange.com/questions/604483/column-delimiter-in-a-macro-to-render-rows-for-tabularray

**Reply by cdacl** [reply]
I temporarily use `\expandafter` to solve this problem. Thank you!
